pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -294,6 +294,13 @@ func translateRoute(in *networking.HTTPRoute,
 		return nil
 	}
 
+	path := getMatchPath(match)
+	operation := fmt.Sprintf("%s:%d%s", vsName, port, path)
+	// use host:port + match uri for operation name if there is only one route.
+	if len(in.Route) == 1 {
+		operation = fmt.Sprintf("%s:%d%s", in.Route[0].Destination.Host, port, path)
+	}
```

Comment:
This logic needs to be in the service level routes as well. Also, what happens if the virtual service name is different from the service name ? The sender and receiver will have differently named spans.

Reply:
Sorry, new to pilot model, what do you mean by service level routes? HttpRouteConfig?

Right now operationa are already different in outbound routes and inbound routes: outbound routes use virtual service name while inbound routes use "default-route". Envoy includes x-envoy-decorator-operation header in egress request, which will override operation name in inbound routes: envoyproxy/envoy/issues/1849

Reply:
Can we remove inbound route decorator? "default-route" is not providing any value. Does inbound decorator override the span decorator initiated from the server?

Reply:
No it is actually the opposite, outbound decorator will override inbound decorator. But anyway I removed default-route from inbound route.